code/clue.js

```diff
@@ -1,59 +1,227 @@
   // This will randomly select a killer from the suspects. And add that to the mystery object.
   mystery.killer = randomSelector(suspects)
+  
+  //This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
+  document.getElementById('killerCard').style.background = mystery.killer.color;
+  document.getElementById('killerImage').src = mystery.killer.image;
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+}
+
+const pickWeapon = () => {
+  // This will randomly select a weapon. And add that to the mystery object.
+  mystery.weapons = randomSelector(weapons)
+
+  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
+  document.getElementById('weaponCard').style.background = mystery.weapons.color
+  document.getElementById('weaponName').innerHTML = `${mystery.weapons.name} <br> Weight:${mystery.weapons.weight}`
+}
+const pickRoom = () => {
+  // This will randomly select a killer from the suspects. And add that to the mystery object.
+  mystery.room = randomSelector(rooms)
 
   // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('roomCard').style.background = mystery.room.color
+  document.getElementById('roomName').innerHTML = `${mystery.room}`
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
 
 // STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+const theMystery = document.getElementById('mystery');
+const revealButton = document.getElementById (`revealButton`);
+const mysteryAnswer = document.getElementById(`mystery`)
+
+const revealMystery = () => {
+  if (mystery.killer.firstName === undefined
+    || mystery.weapons.name === undefined
+    || mystery.room === undefined) 
+    {
+    mysteryAnswer.innerHTML = `No peeking! Guess and click each card first!`
+    } 
+  else {
+theMystery.innerHTML =` The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName}, 
+in the ${mystery.room} with a ${mystery.weapons.name}`
+  }
+};
+
+document.getElementById(`revealButton`).onclick = revealMystery;
+document.getElementById('killerCard').onclick = pickKiller;
+document.getElementById('weaponCard').onclick = pickWeapon;
+document.getElementById('roomCard').onclick = pickRoom;
```

Comment:
I love that ALL of these are grouped together.  I feel like I know where all of your event handling happens.  

Also, if I did this for myself, I would know exactly where to add a new `onclick` handler, should I create a new one in the future!

Reply:
Thank you, trying to embrace order :)